src/Task.ts

```diff
@@ -70,7 +70,7 @@ export class Task {
     public readonly precedingHeader: string | null;
 
     public readonly tags: string[];
-
+    public readonly subTags: string;
```

Comment:
I am a bit unsure about adding a new field on Task for this - I think as I don't think I understand it.

I'm a bit confused by the name. It says subTags, but I am guess from code elsewhere is that it really means 'sub-tags-of the-global-filter, if there is one, and if it is tag, and not just a simple string like `Todo: `.

How does `subTags` relate to tags?

Its name is plural, but it is a string and not an array. If it is multiple values joined together, what characters are they separated by?

Reply:
Probably deserves a name change, you are right!

It's a string that contains all of the subtags on the global filter. So for example the Task:
```markdown
- [ ] #task/sub/sub2 blah blah blah #extra-tag/here
```

`Task.subTags` would be `/sub/sub2`.

I think the explanation on the regex piece will help clarify. Also maybe the name of this field should be `globalFilterSubTags`? (I was actually just playing the modal as well and the naming there could use some clarification as well.

Reply:
I'd like to think about this a bit more.

An example for the tests will be having multiple occurrences of the global filter in a task that is parsed, for example  this (Where global filter is `#task`)

`- [ ] #task/sub/sub2 blah blah blah #task/sub/sub3 #task/sub2/sub`